scripts/extensions/sams/src/api/assets.ts

```diff
@@ -426,3 +426,20 @@ export function getAssetsCompressedBinary(asset_ids: Array<string>): Promise<voi
             return Promise.reject(error);
         });
 }
+
+export function deleteAsset(item: IAssetItem): Promise<string> {
+    const {gettext} = superdeskApi.localization;
+    const {notify} = superdeskApi.ui;
+
+    return superdeskApi.dataApi.delete<IAssetItem>(RESOURCE, item)
+        .then(() => {
+            notify.success(gettext('Asset deleted successfully'));
+
+            return Promise.resolve('Success');
```

Comment:
There shouldn't be a need to return a value inside this `Promise`. When processing the response, if the `Promise` resolved successfully, then continue on as expected. React to errors when `Promise.reject` is returned